src/arrayMethodJoin.js

```diff
@@ -4,9 +4,37 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let separatorString;
+    let finalString = '';
+
+    switch (separator) {
+      case null: separatorString = 'null';
+        break;
+      case undefined: separatorString = 'undefined';
+        break;
+      case '': separatorString = '';
+        break;
+      default: separatorString = separator;
+    }
+
+    for (const element of this) {
+      switch (element) {
+        case undefined: finalString += '';
+          break;
+        case null: finalString += '';
+          break;
+        case NaN: finalString += 'NaN';
+          break;
+        default: finalString += element;
```

Comment:
Same here. You don't need to add empty string and NaN manually. Those values will be added anyway